java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -2093,20 +2081,15 @@ public ExpressionTree completeCreatorSelector(AstNode dotTokenAstNode, Expressio
   public ExpressionTree newDotClassSelector(Optional<List<Tuple<AstNode, AstNode>>> dimensions, AstNode dotTokenAstNode, AstNode classTokenAstNode) {
     IdentifierTreeImpl identifier = new IdentifierTreeImpl(InternalSyntaxToken.create(classTokenAstNode));
 
+    ArrayTypeTreeImpl nestedAnnotations = newArrayTypeTree(dimensions);
     List<AstNode> children = Lists.newArrayList();
-
-    if (dimensions.isPresent()) {
-      for (Tuple<AstNode, AstNode> dimension : dimensions.get()) {
-        children.add(dimension.first());
-        children.add(dimension.second());
-      }
+    if (nestedAnnotations != null) {
+      children.add(nestedAnnotations);
     }
-
     children.add(dotTokenAstNode);
     children.add(identifier);
 
-    return new MemberSelectExpressionTreeImpl(dimensions.isPresent() ? dimensions.get().size() : 0, identifier,
-      children);
+    return new MemberSelectExpressionTreeImpl(nestedAnnotations, identifier, children);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Null passed for non-null parameter of new org.sonar.java.model.expression.MemberSelectExpressionTreeImpl(JavaTree$ArrayTypeTreeImpl, IdentifierTreeImpl, List) in org.sonar.java.ast.parser.TreeFactory.newDotClassSelector(Optional, AstNode, AstNode) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_PARAM_DEREF)